src/main/java/com/mraof/minestuck/item/weapon/MusicPlayerItem.java

```diff
@@ -0,0 +1,194 @@
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.network.NetworkHooks;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Random;
+
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID)
+public class MusicPlayerItem extends WeaponItem
+{
+	private final float volume;
+	private final float pitch;
+	private static IItemHandler getItemStackHandlerMusicPlayer(ItemStack itemStack)
+	{
+		return itemStack.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY).orElseThrow(() ->
+				new IllegalArgumentException("Expected an item handler for the music player item, but " + itemStack + " does not expose an item handler."));
+	}
+	
+	private static IMusicPlaying getMusicPlaying(LivingEntity entity)
+	{
+		return entity.getCapability(CapabilityMusicPlaying.MUSIC_PLAYING_CAPABILITY).orElseThrow(() ->
+				new IllegalArgumentException("Expected an music playing for this entity, but " + entity + " does not expose a music playing."));
+	}
+	
+	@Nullable
+	@Override
+	public ICapabilityProvider initCapabilities(ItemStack stack, @Nullable CompoundTag nbt)
+	{
+		return new MusicPlayerItemCapProvider();
+	}
+	
+	public MusicPlayerItem(Builder builder, Properties properties, float volume, float pitch)
+	{
+		super(builder, properties);
+		this.volume = volume;
+		this.pitch = pitch;
+		
+	}
+	
+	public MusicPlayerItem(Builder builder, Properties properties)
+	{
+		super(builder, properties);
+		this.volume = 4.0F;
+		this.pitch = 1.0F;
+	}
+	
+	@Override
+	public InteractionResultHolder<ItemStack> use(Level level, Player playerIn, InteractionHand handIn)
+	{
+		ItemStack musicPlayer = playerIn.getItemInHand(handIn);
+		
+		IItemHandler itemStackHandlerMusicPlayer = getItemStackHandlerMusicPlayer(musicPlayer);
+		IMusicPlaying musicPlayingCap = getMusicPlaying(playerIn);
+		
+		if(!level.isClientSide)
+		{
+			if(playerIn.isCrouching())
+			{
+				Item itemInMusicPlayer = itemStackHandlerMusicPlayer.getStackInSlot(0).getItem();
+				if(itemInMusicPlayer instanceof CassetteItem cassette)
+				{
+					MusicPlayerPacket packet;
+					if(musicPlayingCap.getCassetteType() == EnumCassetteType.NONE)
+					{
+						packet = MusicPlayerPacket.createPacket(playerIn, cassette.cassetteID, volume, pitch);
+						musicPlayingCap.setMusicPlaying(musicPlayer, cassette.cassetteID);
+					} else
+					{
+						packet = MusicPlayerPacket.createPacket(playerIn, EnumCassetteType.NONE, 0, 0);
+						musicPlayingCap.setMusicPlaying(ItemStack.EMPTY, EnumCassetteType.NONE);
+					}
+					MSPacketHandler.sendToTrackingAndSelf(packet, playerIn);
+				}
+			}
```

Comment:
Looked some more at the problem, and there seems to be two aspects to it:
1. If the music is stopped immediately after it was set to start, the sound engine doesn't seem to register the stop call.  This can happen if there's a packet for playing music immediately followed by another music packet for the same player. This is a problem in `PlayerMusicClientHandler` and need to be fixed there. That class believes that it stopped the music from the first packet during the second packet, but the sound engine didn't actually stop it from playing. Solving this may involve figuring out exactly why the sound engine didn't handle the call to stop the music. It's possible that the end solution may involve playing and stopping music in bulk during some tick event instead of one by one through the execute function of the packet.
2. We're getting a "play music" and a "stop music" packet in the same tick! This is probably because the first music player item doesn't block interaction with the second when used. That's because this function is not returning the proper result when some sort of interaction is attempted or succeeds. This part is the most important one to fix!